sdk/rust/arvados-client/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+
+use serde_json::Value;
+use std::collections::HashMap;
+use serde::{Serialize, Deserialize};
+use std::rc::Rc;
+use std::fmt::{Write};
+use reqwest::{Client};
+use reqwest::header::{HeaderValue, HeaderMap, AUTHORIZATION};
+
+
+// This code is generated by the build.rs script and avados-api-generator.
+include!(concat!(env!("OUT_DIR"), "/arvados-api.rs"));
+
+#[derive(Debug)]
+pub struct ArvadosApi {
+    // Ref counted pointer to client data. Ideally should be a reference.
+    client: Rc<ArvadosClient>,
+}
+
+#[derive(Debug)]
+pub struct ArvadosClient {
+    /// Reqwest client. Could be hyper?
+    http_client: Client,
+
+    /// base of the http request string.
+    base_url: String,
+}
+
+impl ArvadosApi {
+    pub fn new(arv_api_host: &str, arv_api_token: &str, arv_api_host_insecure: bool) -> Result<Self> {
+        Ok(ArvadosApi { client: Rc::new(ArvadosClient::new(arv_api_host, arv_api_token, arv_api_host_insecure)?)})
+    }
+}
+
+type ArvadosError = Box<dyn std::error::Error + Send + Sync>;
+pub type Result<T> = std::result::Result<T, ArvadosError>;
+
+
+impl ArvadosClient {
+    pub fn new(arv_api_host: &str, arv_api_token: &str, arv_api_host_insecure: bool) -> Result<Self> {
+        let proto = if arv_api_host_insecure { "http" } else {"https" };
```

Comment:
Not quite right.  The API server is always https.  The "insecure" flag means you want to accept self-signed certificates.

Reply:
Done. But I've kept http for testing as https is a bit over the top.